crates/bevy_camera_controller/src/free_cam.rs

```diff
@@ -1,37 +1,67 @@
+//! They are rarely useful as-is for gameplay,
+//! as they allow the user to move freely in all directions,
+//! which can be disorienting, and they can clip through objects and terrain.
+//!
+//! You may have heard of a "fly cam" before,
+//! which are a kind of free cam designed for fluid "flying" movement and quickly surveying large areas.
+//! By contrast, the default settings of this particular free cam are optimized for precise control.
+//!
+//! To use this controller, add [`FreeCamPlugin`] to your app,
+//! and [`FreeCam`] to your camera entity.
+//!
+//! To configure the settings of this controller, modify the fields of the [`FreeCam`] component.
 
-use bevy::{
-    input::mouse::{AccumulatedMouseMotion, AccumulatedMouseScroll, MouseScrollUnit},
-    prelude::*,
-    window::{CursorGrabMode, CursorOptions},
+use bevy_app::{App, Plugin, RunFixedMainLoop, RunFixedMainLoopSystems};
+use bevy_camera::Camera;
+use bevy_ecs::prelude::*;
+use bevy_input::keyboard::KeyCode;
+use bevy_input::mouse::{
+    AccumulatedMouseMotion, AccumulatedMouseScroll, MouseButton, MouseScrollUnit,
 };
-use std::{f32::consts::*, fmt};
+use bevy_input::ButtonInput;
+use bevy_log::info;
+use bevy_math::{EulerRot, Quat, Vec2, Vec3};
+use bevy_time::{Real, Time};
+use bevy_transform::prelude::Transform;
+use bevy_window::{CursorGrabMode, CursorOptions, Window};
+
+use core::{f32::consts::*, fmt};
 
 /// A freecam-style camera controller plugin.
-pub struct CameraControllerPlugin;
+///
+/// Use [`FreeCam`] to add a freecam controller to a camera entity,
+/// and change its values to customize the controls and change its behavior.
+pub struct FreeCamPlugin;
 
-impl Plugin for CameraControllerPlugin {
+impl Plugin for FreeCamPlugin {
     fn build(&self, app: &mut App) {
-        app.add_systems(Update, run_camera_controller);
+        // This ordering is required so that both fixed update and update systems can see the results correctly
+        app.add_systems(
+            RunFixedMainLoop,
+            run_freecam_controller.in_set(RunFixedMainLoopSystems::BeforeFixedMainLoop),
+        );
     }
 }
 
+/// Scales mouse motion into yaw/pitch movement.
+///
 /// Based on Valorant's default sensitivity, not entirely sure why it is exactly 1.0 / 180.0,
-/// but I'm guessing it is a misunderstanding between degrees/radians and then sticking with
+/// but we're guessing it is a misunderstanding between degrees/radians and then sticking with
 /// it because it felt nice.
-pub const RADIANS_PER_DOT: f32 = 1.0 / 180.0;
+const RADIANS_PER_DOT: f32 = 1.0 / 180.0;
 
-/// Camera controller [`Component`].
+/// Freecam controller settings and state.
+///
+/// Add this component to a [`Camera`] entity and add [`FreeCamPlugin`]
+/// to your [`App`] to enable freecam controls.
 #[derive(Component)]
-pub struct CameraController {
-    /// Enables this [`CameraController`] when `true`.
+pub struct FreeCam {
+    /// Enables this [`FreeCamController`] when `true`.
```

Comment:
I think all these doc links need updating
```suggestion
    /// Enables this [`FreeCam`] when `true`.
```